### PR TITLE
feat(api): add correlationID to Track Chat requests

### PR DESCRIPTION
**Pull request by siddimore**
**Description**

This PR does not fix but a feature to add correlationID to track chat requests

1. Add CorrelationID to chat completions endpt
2. Pass this CorrelationID to backend grpc-cpp server
3. Log CorrelationID to help track requests

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: core/http/endpoints/openai/chat.go [hunk 1]
```diff
@@ -161,6 +161,12 @@ func ChatEndpoint(cl *config.BackendConfigLoader, ml *model.ModelLoader, startup
 		textContentToReturn = ""
 		id = uuid.New().String()
 		created = int(time.Now().Unix())
+		// Set CorrelationID
+		correlationID := c.Get("X-Correlation-ID")
+		if len(strings.TrimSpace(correlationID)) == 0 {
+			correlationID = id
+		}
+		c.Set("X-correlationID", correlationID)
```
Comment (mudler) [inline core/http/endpoints/openai/chat.go:169]: Is this a typo? 

```suggestion
		c.Set("X-Correlation-ID", correlationID)
```

Comment (siddimore) [reply core/http/endpoints/openai/chat.go:169]: good catch it is fixing

File: core/http/endpoints/openai/chat.go [hunk 2]
```diff
@@ -444,6 +450,7 @@ func ChatEndpoint(cl *config.BackendConfigLoader, ml *model.ModelLoader, startup
 			c.Set("Cache-Control", "no-cache")
 			c.Set("Connection", "keep-alive")
 			c.Set("Transfer-Encoding", "chunked")
+			c.Set("X-correlationID", id)
```
Comment (mudler) [inline core/http/endpoints/openai/chat.go:453]:
```suggestion
			c.Set("X-Correlation-ID", id)
```

**Comment by mudler** [comment]
@siddimore This looks good here!

I need DCO to be cleared out before merging - can you sign-off your commits with `git rebase origin/master --signoff` ?

**Comment by siddimore** [comment]
> @siddimore This looks good here!
> 
> I need DCO to be cleared out before merging - can you sign-off your commits with `git rebase origin/master --signoff` ?

yup @mudler will do it today. I also have another PR in the wings for exposing tokens_per_second as a metric. Will raise it with signed off commits.

**Review by mudler — APPROVED** [review]

**Comment by siddimore** [comment]
@mudler had to rebase please kindly re-approve

**Review by mudler — APPROVED** [review]

**Comment by siddimore** [comment]
@mudler  thanks, i am PST TZ watching the build lastnight until 11:30pm noticed the merge.

**Pull request merged**